### DISCUSSION TITLE
How to force uvicorn to print logs in JSON

### DISCUSSION THREAD

**Discussion by vveliev-tc** [fastapi/fastapi#7726] [Questions]
### Description

How can I configure uvicorn main logger  to print all application messages as JSON using  [aiologger](https://github.com/b2wdigital/aiologger) 

### Additional context
tried editing . gunicorn_conf.py
```py
import code.logging
import json
import multiprocessing
import os

from aiologger.loggers.json import JsonLogger

logger = JsonLogger.with_default_handlers()



workers_per_core_str = os.getenv("WORKERS_PER_CORE", "1")
web_concurrency_str = os.getenv("WEB_CONCURRENCY", None)
host = os.getenv("HOST", "0.0.0.0")
port = os.getenv("PORT", "80")
bind_env = os.getenv("BIND", None)
use_loglevel = os.getenv("APP_LOG_LEVEL", "info")
if bind_env:
    use_bind = bind_env
else:
    use_bind = f"{host}:{port}"

cores = multiprocessing.cpu_count()
workers_per_core = float(workers_per_core_str)
default_web_concurrency = workers_per_core * cores
if web_concurrency_str:
    web_concurrency = int(web_concurrency_str)
    assert web_concurrency > 0
else:
    web_concurrency = max(int(default_web_concurrency), 2)

# Gunicorn config variables
loglevel = use_loglevel
workers = web_concurrency
bind = use_bind
keepalive = 120
errorlog = "-"

# For debugging and testing
log_data = {
    "loglevel": loglevel,
    "workers": workers,
    "bind": bind,
    # Additional, non-gunicorn variables
    "workers_per_core": workers_per_core,
    "host": host,
    "port": port,
}
print(json.dumps(log_data))

```

**Comment by dmontagu** [comment]
Unfortunately I don't know enough to help here, but you might also try asking in the uvicorn repo, as this question isn't really specific to fastapi in any way.

**Comment by nkhitrov** [comment]
Hi, @vveliev-tc !

Perhaps no longer relevant, but you can create custom handler for uvicorn logger to parse messages into json or something else. 

Here [example](https://github.com/tiangolo/fastapi/issues/1276#issuecomment-615877177) with loguru (not json format).

**Comment by riki-nitdgp** [comment]
Try this, will help you https://medium.com/1mgofficial/how-to-override-uvicorn-logger-in-fastapi-using-loguru-124133cdcd4e

Change log settings

**Comment by ccrvlh** [comment]
@dmontagu @tiangolo suggestion to close this issue, as this seems specifically related to Uvicorn/aiologger.
